### ISSUE TITLE
How to define nested list of ints?

### ISSUE THREAD

**Issue by michalkvasnicak** [hayes/pothos#662]
We have a field of `[[Int!]!]!` type but I couldn't find a way how to define that. Also types don't seem to allow that either as only the typing is `export type TypeParam<Types extends SchemaTypes> = OutputType<Types> | [OutputType<Types>];` which support only single level list.

Are nested lists supported by Pothos?

**Comment by michalkvasnicak** [comment]
@hayes thank you! I was thinking about it when I was trying to find out whether this is somehow supported but not documented and I explored things like nested arrays, for example `[['Int']]` but then I realized that it would be hard to tell about `nullability`. So then I tried something like `[t.field({ nullable: { list: false, items: false }, type: ['Int']])]`. These 2 things were like natural to me. 

So if that might help (at least with an idea), perhaps there could be some type builder exposed so we would be able to define nested lists, for example:

```
builder.objectType('A', {
  fields: t => ({
     list: t.list({
       nullable: { list: false, items: false },
       type: t.list({ type: 'Int', nullable: { list: false, items: false }}),
     }),
  }),
})
```
